### PR TITLE
Nimbus+ mac HID support

### PR DESCRIPTION
**Pull request by ekcoh**
### Description

Adds support for SteelSeries Nimbus+ gamepad on OSX where the device is surfaced as a HID device. (Contribution by [MollyJameson](https://github.com/MollyJameson)).
It should be noted that this device is enumerated with bogus Vendor and Product ID.

### Changes made

Original contribution:
- Gamepad layout mapping. See https://github.com/Unity-Technologies/InputSystem/pull/1318/files

Additional changes added to the original contribution:
- Renamed "menuButton" to "homeButton" since this button is referenced as "Home Button" in the device manual: https://steelseries.com/gaming-controllers/nimbus-plus 
- Added unit tests.
- Added more string matching against OSX VID/PID provided via HID and regexp turned off.

### Notes

Note that this doesn't add support for SteelSeries Nimbus.
Note that the device is not picked up by e.g. Windows when using the provided vendor and product name. The original contribution could be picked up on other OS when using proper product and vecor ID. 

### Checklist

Before review:

- [ ] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/Plugins/OSX/OSXGameController.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        [FieldOffset(7)] public byte buttons2;
+    }
+}
+
+namespace UnityEngine.InputSystem.OSX
+{
+    /// <summary>
+    /// Steel Series Nimbus+ uses iOSGameController MFI when on iOS but
+    /// is just a standard HID on OSX. Note that the gamepad is made available
+    /// with incorrect VID/PID by OSX. Instead of
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/Plugins/OSX/OSXGameController.cs]: Unfinished sentence.

**Review by Rene-Damm — APPROVED** [review]

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +40,9 @@ however, it has to be formatted properly to pass verification tests.
 
 - Fixed `InputAction.GetTimeoutCompletionPercentage` jumping to 100% completion early ([case 1377009](https://issuetracker.unity3d.com/issues/gettimeoutcompletionpercentage-returns-1-after-0-dot-1s-when-hold-action-was-started-even-though-it-is-not-performed-yet)).
 
+### Added
+- Added support for SteelSeries Nimbus+ gamepad on Mac via HID and explicit layout (Addition contributed by [Mollyjameson](https://github.com/MollyJameson)).
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Maybe just "Added support for SteelSeries Nimbus+ gamepad on Mac" without the "via HID and explicit layout" part (seems like mere implementation detail).

**Review by jimon — APPROVED** [review]

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 2 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Good point, changed

File: Packages/com.unity.inputsystem/InputSystem/Plugins/OSX/OSXGameController.cs (hunk 1 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/InputSystem/Plugins/OSX/OSXGameController.cs]: Thanks, fixed

**Pull request merged**